src/main/java/com/squareup/javapoet/Util.java

```diff
@@ -109,4 +110,19 @@ public static void requireExactlyOneOf(Set<Modifier> modifiers, Modifier... mutu
   public static boolean hasDefaultModifier(Collection<Modifier> modifiers) {
     return DEFAULT != null && modifiers.contains(DEFAULT);
   }
+
+  public static String identifier(String part) {
```

Comment:
I revoke my last two comments.

            .addMethod(MethodSpec.constructorBuilder()
                .addParameter(Thread.State[].class, "states")
                .varargs(true)
                .build())

produces your prediction

	java.lang.IllegalArgumentException: part[0] is not valid
		at com.squareup.javapoet.Util.checkArgument(Util.java:63)
		at com.squareup.javapoet.Util.identifier(Util.java:117)
		at com.squareup.javapoet.CodeWriter.emit(CodeWriter.java:271)